src/Scheduler.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace Rx;
+
+use Rx\Scheduler\EventLoopScheduler;
+use Rx\Scheduler\ImmediateScheduler;
+
+class Scheduler
+{
+    private static $default;
+    private static $async;
+    private static $immediate;
+
+    public static function getDefault(): SchedulerInterface
+    {
+        if (!static::$default) {
+            static::$default = new EventLoopScheduler();
+        }
+
+        return static::$default;
+    }
+
+    public static function setDefault(SchedulerInterface $scheduler)
```

Comment:
The static schedulers should not be settable. If they are - they should only be set once or use a factory.